network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/failure/FailureRecoveryHandler.kt

```diff
@@ -79,7 +90,7 @@ class FailureRecoveryHandler @Inject constructor(
         var currentDelay = initialDelay
         repeat(times) {
             try {
-                if (!recoveryCompleted) {
+                if (failureRecoveryInProgress) {
```

Comment:
as long `failureRecoveryInProgress` is `true`, we continue to attempt recovery (with backoff). `failureRecoveryInProgress` is marked `true` when onTunnelFailure is called and marked `false` when `onTunnelFailureRecovered` is called.